packages/test/src/util/browser-tests-runner/driver.js

```diff
@@ -122,15 +133,45 @@ function connect({ viewport = DEFAULT_VIEWPORT, ...options }, capability) {
       }
     })
     .init()
-    .url("")
-    .timeouts("script", timeout)
-    .timeouts("implicit", timeout)
-    .timeouts("page load", timeout)
+    .timeouts("script", idleTimeout)
+    .timeouts("implicit", idleTimeout)
+    .timeouts("page load", idleTimeout)
     .setViewportSize(viewport)
+    .url("")
     .then(() => () => {
-      return driver.executeAsync(function(done) {
-        window.__run_tests__(done);
-      });
+      const endTime = Date.now() + suiteTimeout;
+      const execInterval = Math.min(idleTimeout, DEFAULT_TIMEOUTS.idle) * 0.8;
+      // Some services force a max timeout for async scripts.
+      // Below we restart the async script the waits for the tests to be done every 60 seconds.
+      // The tests are 'complete' once the global '__test_result__' is set on the window.
+      return (function getResults() {
+        if (endTime < Date.now()) {
+          throw new Error(
+            'marko-cli: Test suite timed out, use "wdioOptions.suiteTimeout" to increase the delay (default 10 mins).'
+          );
+        }
+
+        return driver
+          .executeAsync(function(interval, done) {
+            if (window.__test_result__) done(window.__test_result__);
+            else {
+              var timeout = setTimeout(function() {
+                delete window.__test_result__;
+                done();
+              }, interval);
+              Object.defineProperty(window, "__test_result__", {
+                configurable: true,
+                set: function(result) {
+                  clearTimeout(timeout);
+                  done(result);
```

Comment:
This is unlikely, but I think it may be possible that `done` gets called twice in the unfortunate event that the timeout function actually gets called before `window.__test_result__` is set. Probably worth adding a check here to ensure that `done` has not been called already.

Reply:
I might be missing something but this shouldn't be the case. The timeout cleans up the `__test_result__` getter/setter and so it shouldn't call done again. Once the `executeAsync` happens again the next time we check to see if the `__test_result__` was added (which would have skipped the setter).